core/src/main/java/org/apache/calcite/rex/RexLiteral.java

```diff
@@ -1275,6 +1275,11 @@ public static Number numberValue(RexNode node) {
    * never null. */
   public static int intValue(RexNode node) {
     final Number number = numberValue(node);
+    long longValue = number.longValue();
+    if (longValue > Integer.MAX_VALUE || longValue < Integer.MIN_VALUE) {
+      throw new ArithmeticException(
+          "integer overflow: Value " + longValue + " out of range for INT");
```

Comment:
Nit: ```suggestion
          "Integer overflow: " + longValue + " is out of range for INT");
```?

Reply:
Completed the conversion integer -> Integer